Dịch "anchor_vn" - Phần 4 (#2266)
aivivn/d2l-vn: chapter_computer-vision/anchor_vn.md

At this point, elements in two rows and two columns in the matrix $\mathbf{X}$ have been discarded.
-->

> Giả sử rằng những khung neo trên ảnh là $A_1, A_2, \ldots, A_{n_a}$ và những khung chứa chuẩn là $B_1, B_2, \ldots, B_{n_b}$ and $n_a \geq n_b$.

Reviewer: ```suggestion
Giả sử rằng những khung neo trên ảnh là $A_1, A_2, \ldots, A_{n_a}$ và những khung chứa nhãn gốc là $B_1, B_2, \ldots, B_{n_b}$ and $n_a \geq n_b$.
```

> Định nghĩa ma trận $\mathbf{X} \in \mathbb{R}^{n_a \times n_b}$, trong đó mỗi phần tử $x_{ij}$ trong hàng $i^\mathrm{th}$ và cột $j^\mathrm{th}$ là hệ số IoU của khung neo $A_i$ so với khung chứa chuẩn $B_j$.

Reviewer: ```suggestion
Định nghĩa ma trận $\mathbf{X} \in \mathbb{R}^{n_a \times n_b}$, trong đó mỗi phần tử $x_{ij}$ trong hàng $i^\mathrm{th}$ và cột $j^\mathrm{th}$ là hệ số IoU của khung neo $A_i$ so với khung chứa nhãn gốc $B_j$.
```

Đầu tiên, chúng ta tìm ra phần tử lớn nhất trong ma trận $\mathbf{X}$ rồi lưu lại chỉ mục hàng và cột của phần tử đó là $i_1,j_1$.

> Chúng ta gán khung chứa chuẩn $B_{j_1}$ cho khung neo $A_{i_1}$.

Reviewer: ```suggestion
Chúng ta gán khung chứa nhãn gốc $B_{j_1}$ cho khung neo $A_{i_1}$.
```

> Ta thấy rõ ràng rằng, khung neo $A_{i_1}$ và khung chứa chuẩn $B_{j_1}$ có độ tương đồng cao nhất trong số tất cả các cặp "khung neo--khung chứa chuẩn".

Reviewer: ```suggestion
Ta thấy rõ ràng rằng, khung neo $A_{i_1}$ và khung chứa nhãn gốc $B_{j_1}$ có độ tương đồng cao nhất trong số tất cả các cặp "khung neo--khung chứa nhãn gốc".
```

Tiếp theo, loại bỏ các phần tử trong hàng $i_1$th và cột $j_1$th trong ma trận $\mathbf{X}$.

> Tìm ra phần tử lớn nhất còn lại trong ma trận $\mathbf{X}$ rồi cũng lưu lại chỉ mục hàng và cột của phần tử đó là $i_2,j_2$.

Reviewer: ```suggestion
Tìm ra phần tử lớn nhất trong các phần tử còn lại trong ma trận $\mathbf{X}$ rồi cũng lưu lại chỉ mục hàng và cột của phần tử đó là $i_2,j_2$.
```

> Chúng ta gán khung chứa chuẩn $B_{j_2}$ cho khung neo $A_{i_2}$ và sau đó loại bỏ mọi phần tử tại hàng $i_2$th và cột $j_2$th trong ma trận $\mathbf{X}$.

Reviewer: ```suggestion
Chúng ta gán khung chứa nhãn gốc $B_{j_2}$ cho khung neo $A_{i_2}$ và sau đó loại bỏ mọi phần tử tại hàng $i_2$th và cột $j_2$th trong ma trận $\mathbf{X}$.
```

> Như vậy, tại thời điểm này thì các phần tử trong hai hàng và hai cột vừa rồi của ma trận $\mathbf{X}$ đã bị loại bỏ.

Reviewer: ```suggestion
Như vậy, tại thời điểm này thì các phần tử trong hai hàng và hai cột của ma trận $\mathbf{X}$ đã bị loại bỏ.
```

and only assign ground-truth bounding box $B_j$ to anchor box $A_i$ when the IoU is greater than the predetermined threshold.
-->

Chúng ta tiến hành việc này cho đến khi các phần tử ở cột $n_b$ trong ma trận $\mathbf{X}$ đều bị loại bỏ.

> Tại thời điểm này, chúng ta đều đã gán một khung chứa chuẩn cho mỗi khung neo $n_b$.

Reviewer: ```suggestion
Tại thời điểm này, chúng ta đều đã gán $n_b$ khung chứa nhãn gốc cho $n_b$ khung neo.
```
thế này rõ nghĩa hơn.

> Tiếp đến, chúng ta chỉ việc duyệt qua các khung neo $n_a - n_b$ còn lại.

Reviewer: ```suggestion
Tiếp đến, chúng ta chỉ việc duyệt qua $n_a - n_b$ khung neo còn lại.
```

> Nhận đầu vào là khung neo $A_i$, ta cần tìm ra khung chứa $B_j$ sao cho khung chứa ấy có hệ số IoU so với $A_i$ là giá trị lớn nhất, dựa trên mỗi hàng $i^\mathrm{th}$ của ma trận $\mathbf{X}$,

Reviewer: ```suggestion
Với khung neo $A_i$, ta cần tìm ra khung chứa nhãn gốc $B_j$ sao cho khung chứa ấy có hệ số IoU so với $A_i$ là lớn nhất trên mỗi hàng $i^\mathrm{th}$ của ma trận $\mathbf{X}$,
```

> và chỉ gán khung chứa chuẩn $B_j$ cho khung neo $A_i$ khi mà hệ số IoU lớn hơn một ngưỡng đã được định sẵn.

Reviewer: ```suggestion
và chỉ gán khung chứa nhãn gốc $B_j$ cho khung neo $A_i$ khi mà hệ số IoU lớn hơn một ngưỡng cho trước.
```
